sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java

```diff
@@ -1044,9 +1136,9 @@ private void releaseAndOfferChannelIfHealthy(
      * promise} completes with an {@link IllegalStateException}.
      */
     private void releaseChannel(final Channel channel, final Promise<Void> promise) {
-
         checkState(channel.eventLoop().inEventLoop());
 
+        // TODO: moderakh what is ChannelPool
```

Comment:
RntbdClientChannelPool is a ChannelPool. 

So far my understanding is its an abstraction facilitating the pooling mechanism. Haven't seen it coupled with the core netty Channels or handlers. 
The attribute is a way of correlating it with the pool. Below mechanism is to purely ensure that release done on the same pool. 

else case is interesting. Closing on this channel will only release the underlying channel but the associated pool state will be in-consistent. 
Mi-targeting can happen if there are multiple ServcieEndpints for the same remote address, which means idle-time handling. 

Lets not gate these fixes on it. 
Please remove this comment.

Reply:
Agreed. Removed the TODO.